src/server/main.py

```diff
@@ -0,0 +1,370 @@
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>
+
+import time
+import random
+import json
+
+from flask import Flask, request
+from flask_cors import CORS
+
+from src.cvrp.ecvrp import ECVRPSolution, ECVRPInstance
+from src.cvrp.json_io import JsonWriter, read_json, get_header
+from src.cvrp.ga import GA
+from src.cvrp import constraints_validators
+from .utils import utils
+
+HYPER_LIST = {
+    "ga": [
+        "nb_epochs",
+        "pop_size",
+        "mutation_rate",
+        "crossover_rate"
+    ]
+}
+
+
+def build_first_gen(size: int, instance: ECVRPInstance):
+    """Build a first geenration of n valid individuals."""
+    validators = [
+        constraints_validators.BatteryTWValidator(),
+        constraints_validators.CapacityValidator(),
+        constraints_validators.VehiculeCountValidator()
+    ]
+
+    towns = instance.get_towns()
+
+    first_gen: list[ECVRPSolution] = []
+
+    while len(first_gen) < size:
+        solution = [*towns]
+        random.shuffle(solution)
+
+        cum_dem: int = 0
+        insert_points: list[int] = []
+
+        for i, point in enumerate(solution):
+            cum_dem += instance.get_demand(point)
+            if cum_dem > instance.get_ev_capacity():
+                insert_points.append(i-1 + len(insert_points))
+                cum_dem = 0
+
+        if len(insert_points) > instance.get_ev_count():
+            continue
+
+        for point in insert_points:
+            solution.insert(point, instance.get_depot())
+
+        solution.insert(0, instance.get_depot())
+        solution.append(instance.get_depot())
+        element = ECVRPSolution(validators, solution, instance)
+
+        element.validate()
+
+        if element.is_valid():
+            first_gen.append(element)
+
+    return first_gen
+
+
+class Server:
+    """This class holds the server, its state and routes."""
+
+    def __init__(self, name: str) -> None:
+        """
+        Initialize the server. Does not start it.
+
+        :param name: The name of the Flask application
+        """
+        self._runner = None
+        self._snapshot = None
+        self._nb_it = 0
+        self._count = 0
+        self._tot_time = 0
+        self._override = True
+        self._name = ""
+        self._rate = 0
+
+        self.app = Flask(name)
+        # app = Flask(__name__, static_url_path='', static_folder='react_client/build')
```

Comment:
Why did you let this comment line? 
Is it for a specific case ?

Reply:
It is planned to serve the html, css etc files with the flask server. This line does this. That's why I keep this line.

Reply:
Thank you for the precision